Notifications/Inbox

ColinRyan03 opened the pull request:
Stories
===

*As a manager, I want to receive notifications about upcoming due dates for grants so that we can meet all deadlines.*

*As a manager, I want to receive a notification when a grant is open to start applying so that nothing gets forgotten.*

Acceptance Criteria:
---
1. Notifications for grant open and due dates are displayed in the inbox
2. Notifications can be clicked on to reveal further information

Log:
-Created createNotification.php file
-Edited base.css for create notif button
-Added create-notif functions in event.css
-Created inbox.css
-Fixed errors in dbEvents.php
-Edited dbmessages.php to reflect new db changes and functionality -Edited inbox.php to reflect new db changes and functionality -Edited event.js for creation notif button/pop-up window -Edited viewNotification.php so you can view notif details
-Edited .sql file, changing dbMessages

ColinRyan03 committed:
Notifications/Inbox

-Created createNotification.php file
-Edited base.css for create notif button
-Added create-notif functions in event.css
-Created inbox.css
-Fixed errors in dbEvents.php
-Edited dbmessages.php to reflect new db changes and functionality
-Edited inbox.php to reflect new db changes and functionality
-Edited event.js for creation notif button/pop-up window
-Edited viewNotification.php so you can view notif details
-Edited .sql file, changing dbMessages

colbypexton commented on viewNotification.php:
```diff
@@ -55,8 +55,8 @@
             <span><label>Received </label><?php 
-                    $unpackedTimestamp = unpackMessageTimestamp($message['time']);
-                    echo $unpackedTimestamp[0] . ' at ' . $unpackedTimestamp[1];
+                    $time = $message['time'];
```
Consider renaming to time to timeRecieved for better readability

ColinRyan03 replied:
Good idea, will do

colbypexton commented on js/event.js:
```diff
@@ -57,4 +72,27 @@ $(function() {
+    // from http://stackoverflow.com/questions/45888/what-is-the-most-efficient-way-to-sort-an-html-selects-options-by-value-while
+    var my_options = $('.facilities select option');
+    var selected = $('.facilities').find('select').val();
+  
```
Could you add a comment that briefly describes this section?

ColinRyan03 replied:
This function is actually from a previous bit of code that I was working with that is now deleted. I believe it's no longer necessary, that's probably why it doesn't make sense in its context

colbypexton replied:
Got it. Thanks for clarifying!

colbypexton commented on inbox.php:
```diff
@@ -20,26 +20,77 @@
         <title>Downtown Greens | Inbox</title>
     </head>
     <body>
+        <?php if ($accessLevel >= 2) : ?>
```
Can we confirm that access level is still necessary

ColinRyan03 replied:
I will have to discuss with others in the group if we are considering changing the access levels, or ask which access levels are assigned to which roles (admin, manager, etc.). It might not be necessary

colbypexton replied:
Got it. Let's discuss at the next meeting

ColinRyan03 committed:
Merge branch 'main' into notifications

ColinRyan03 committed:
Bug fixes for notifications

-New class in messages.css for anchors in viewNotification.php
-viewNotification.php cleaned up
-dbMessages.php bug fixes and new functions
-deleteNotification.php updated to redirect correctly
-inbox.php changed to display if message was read yet or not
-function in event.js from last commit deleted (no longer needed)
